### PR TITLE
Feature/std class extension

### PR DESCRIPTION
**Pull request by SimonRubenDrauz**
making std class more general and integrating interpolation std class

### CODE REVIEW

File: pandapipes/std_types/std_type_class.py [hunk 1]
```diff
@@ -35,30 +39,131 @@ def from_dict(cls, d):
...
+    @classmethod
+    def _from_path(cls, path):
+        """
+
+        :param name: Name of the pump object
+        :type name: str
+        :param path: Path where the CSV file, defining a pump object, is stored
+        :type path: str
+        :return: An object of the pump standard type class
+        :rtype: PumpStdType
```
Comment (jkisse) [inline pandapipes/std_types/std_type_class.py:137]: This docstring seems outdated

File: pandapipes/std_types/std_type_class.py [hunk 2]
```diff
@@ -67,88 +172,78 @@ def get_pressure(self, vdot_m3_per_s):
         assumed.
 
         :param vdot_m3_per_s: Volume flow rate of a fluid in [m^3/s]. Abs() will be applied.
```
Comment (jkisse) [inline pandapipes/std_types/std_type_class.py:174]: As far as I understand it, abs(vdot_m3_per_s) is not applied. Instead, bypassing is assumed for reverse flows

Comment (SimonRubenDrauz) [reply pandapipes/std_types/std_type_class.py]: Removed it as your point was already explained in the sentence before.

File: pandapipes/std_types/std_type_class.py (hunk 1 above)
Comment (SimonRubenDrauz) [reply pandapipes/std_types/std_type_class.py]: You are right. Adapted it accordingly.

File: pandapipes/std_types/std_type_class.py [hunk 3]
```diff
@@ -35,120 +39,212 @@ def from_dict(cls, d):
...
+    def load_data(cls, path):
+        raise NotImplementedError
+
+
+class RegressionStdType(StdType):
+
+    def __init__(self, name, component, reg_par):
         """
-        super(PumpStdType, self).__init__(name, 'pump')
+
```
Comment (jkisse) [inline pandapipes/std_types/std_type_class.py:102]: A basic description of this class would be helpful, also to differentiate it from the InterpolationStdType

File: pandapipes/std_types/std_type_class.py [hunk 4]
```diff
@@ -35,120 +39,212 @@ def from_dict(cls, d):
...
-    :param path:
-    :type path:
+    :param p_values:
+    :type p_values:
+    :param v_values:
+    :type v_values:
+    :param degree:
+    :type degree:
     :return:
     :rtype:
```
Comment (jkisse) [inline pandapipes/std_types/std_type_class.py:239]: docstring needs to be updated

File: pandapipes/std_types/std_type_class.py [hunk 5]
```diff
@@ -35,120 +39,212 @@ def from_dict(cls, d):
...
+    if not int(degree) == degree:
+        raise UserWarning("The polynomial degree has to be an integer, but %s was given. "
+                          "It will be rounded down now." % str(degree))
+    return np.polyfit(x_values, y_values, degree)
 
 
-def regression_function(p_values, v_values, degree):
+def interpolation_function(x_values, y_values, fill_value='extrapolate'):
     """
     Regression function...
```
Comment (jkisse) [inline pandapipes/std_types/std_type_class.py:249]: docstring is outdated. Wrong parameter names and "regression function" should be "interpolation... "

File: pandapipes/std_types/std_type_class.py [hunk 6]
```diff
@@ -35,120 +39,212 @@ def from_dict(cls, d):
         return obj
 
 
-class PumpStdType(StdType):
+class InterpolationStdType(StdType):
 
-    def __init__(self, name, reg_par):
+    def __init__(self, name, component, int_fct):
         """
```
[the remark this answers is not in the document]
Comment (jkisse) [reply pandapipes/std_types/std_type_class.py:46]: component and int_fct are missing in the docstring, can you please add it @SimonRubenDrauz ?

[the remark this answers is not in the document]
Comment (SimonRubenDrauz) [reply pandapipes/std_types/std_type_class.py:46]: added it

File: pandapipes/std_types/std_type_class.py (hunk 3 above)
Comment (SimonRubenDrauz) [reply pandapipes/std_types/std_type_class.py:102]: added it

File: pandapipes/std_types/std_type_class.py (hunk 5 above)
Comment (SimonRubenDrauz) [reply pandapipes/std_types/std_type_class.py:249]: updated

File: pandapipes/std_types/std_type_class.py (hunk 3 above)
Comment (jkisse) [reply pandapipes/std_types/std_type_class.py:106]: thanks

**Review by jkisse — APPROVED** [review]

**Pull request merged**